forum/qa-plugin/report-reason/q2apro-flag-reasons-page.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+class q2apro_flag_reasons_page
+{
+
+    private $directory;
+    private $urlToRoot;
+
+    public function load_module($directory, $urlToRoot)
+    {
+        $this->directory = $directory;
+        $this->urlToRoot = $urlToRoot;
+    }
+
+    public function suggest_requests()
+    {
+        return [
+            [
+                'title'   => 'Ajax Flagger', // title of page
+                'request' => 'ajaxflagger', // request name
+                'nav'     => 'M', // 'M'=main, 'F'=footer, 'B'=before main, 'O'=opposite main, null=none
+            ],
+        ];
+    }
+
+    public function match_request($request)
+    {
+        return 'ajaxflagger' === $request;
+    }
+
+    public function process_request($request)
+    {
+
+        $logged = qa_is_logged_in();
+        if (!$logged) {
+            exit();
```

Comment:
bez żadnego komunikatu?

Reply:
To jest endpoint pod który idzie request od frontu, nie ma więc sensu generować cały front z pomocą `qa_prepare_content()`, chyba ze chodzi o jakiegos prostego JSONa w stylu `{"message": "You must be logged in to perform this action"}`